game/scripts/vscripts/modifiers/modifier_aura_item_upgrade.lua

```diff
@@ -50,6 +54,9 @@ if IsServer() then
         end
       end
     end
+
+    self:Destroy()
```

Comment:
Was this tested? I would think it would cause the modifier to destroy itself before it has a chance to run its think function.

Reply:
Oh. Duh. It returns early when it finds an item.

Reply:
Without the DestroyOnExpire it was getting destroyed after the first think